### DISCUSSION TITLE
Send an action at application launch?

### DISCUSSION THREAD

**Discussion by sbeitzel** [pointfreeco/swift-composable-architecture#2279] [Q&A] [answered]
I would like to send an, "application has launched," event into the store at startup time, to kick off background data fetching &c., and I'm wondering what's the right way to do that. I've already tried putting it in an `.onAppear` block on my main view, but that seems actually to be executed multiple times (that is, every time the view changes from one thing to another).

**Comment by sbeitzel** [comment]
For the moment, I've stored a boolean on the application's `State` to record whether or not the "application launched" action has come through the reducer. The app's reducer checks the boolean and if it's false, flips it to `true` and fires off all the background fetch requests, otherwise it does nothing. This works, but it feels a little inelegant.

**Comment by maksimn** [comment]
Hello, @sbeitzel 
It is possible to call `ViewStore(store).send(.appLaunched)` in the initializer of the main view. In this case the `appLaunched` action will be called a single time and it allows to start all required requests.

**Reply by acosmicflamingo** [reply, marked as answer]
Wasn't there a PR that was merged recently that allows sending actions directly from a `Store` instance?
```swift
store.send(.appLaunched)
```

**Reply by sbeitzel** [reply]
There was, and I just forgot about it for a moment. Thank you!

**Comment by Nevwild** [comment]
Try putting store.send(.appLaunched) in the App struct’s init. Best,Nevill WilderOn Jul 10, 2023, at 17:00, Cosmic Flamingo ***@***.***> wrote:﻿
Wasn't there a PR that was merged recently that allows sending actions directly from Store?
store.send(.appLaunched)
